### PR TITLE
Fix player pet check

### PR DESCRIPTION
**Pull request by dtw**
Pretty sure that player.pet_index doesn't exist/always returns `nil`? Tested by adding debugs and loading autocontrol with pet already active.

The validation on line 93 is the same as used later.

### CODE REVIEW

**Comment by z16** [comment]
I think you're right, but to get the current player please use one of these:
```lua
windower.ffxi.get_mob_by_index(player.index)
windower.ffxi.get_mob_by_target('me')
```
The problem with `windower.ffxi.get_mob_by_id` is that it has to iterate the array to find the right match, so its performance is much worse than the other two options. If you have the option, `windower.ffxi.get_mob_by_index` is the best choice, and `windower.ffxi.get_mob_by_target` is the next best.

**Comment by dtw** [comment]
OK, I'll fix that case and the other cases in a subsequent commit

File: addons/autocontrol/maneuver.lua [hunk 1]
```diff
@@ -91,10 +91,10 @@ Burden_tb = texts.new(str, settings)
 windower.register_event("action", function(act)
     if mjob_id == 18 then
         local abil_ID = act['param']
-        local actor_id = act['actor_id']
+        local actor_index = act['actor_index']
```
Comment (dtw) [inline addons/autocontrol/maneuver.lua]: Have not check `actor_index` exists but I assume it does.

**Comment by dtw** [comment]
I have not tested this in game yet

File: addons/autocontrol/maneuver.lua (hunk 1 above)
Comment (z16) [reply addons/autocontrol/maneuver.lua]: It does not.

Comment (dtw) [reply addons/autocontrol/maneuver.lua]: Shucks.

**Comment by dtw** [comment]
This doesn't work at all as expected in testing.

**Pull request closed without merging**